src/text/parsers/symbol.rs

```diff
@@ -253,4 +233,14 @@ mod symbol_parsing_tests {
             TextValue::Symbol(text_token(expected)),
         )
     }
+
+    #[test]
+    fn operators_and_negative_numbers() {
+        parse_test_ok(parse_operator, "--3", TextValue::Symbol(text_token("--")));
```

Comment:
🗺️ I confirmed with `ion-java` and `ion-c` that the correct parse for the children of `(--3)` is `--` and `3`.